flocker/acceptance/test_postgres.py

```diff
@@ -68,7 +50,41 @@ def setUp(self, cluster):
 
         self.node_1, self.node_2 = cluster.nodes
 
-        postgres_deployment = {
+        new_dataset_id = unicode(uuid4())
+
+        self.POSTGRES_APPLICATION = Application(
+            name=POSTGRES_APPLICATION_NAME,
+            image=DockerImage.from_string(POSTGRES_IMAGE + u':latest'),
+            ports=frozenset([
+                Port(internal_port=POSTGRES_INTERNAL_PORT,
+                     external_port=POSTGRES_EXTERNAL_PORT),
+                ]),
+            volume=AttachedVolume(
+                manifestation=Manifestation(
+                    dataset=Dataset(
+                        dataset_id=new_dataset_id,
+                        metadata=pmap({"name": POSTGRES_APPLICATION_NAME}),
+                        maximum_size=REALISTIC_BLOCKDEVICE_SIZE),
+                    primary=True),
+                mountpoint=FilePath(POSTGRES_VOLUME_MOUNTPOINT),
+            ),
+        )
+
+        requested_dataset = {
+            u"primary": self.node_1.uuid,
+            u"dataset_id": new_dataset_id,
+            u"maximum_size": REALISTIC_BLOCKDEVICE_SIZE,
+            u"metadata": {u"name": POSTGRES_APPLICATION_NAME},
+        }
+
+        configuring_dataset = cluster.create_dataset(requested_dataset)
+        configuring_dataset.addCallback(
+            lambda (cluster, dataset): cluster.wait_for_dataset(dataset)
+        )
```

Comment:
These two lines make the test less realistic; normal flocker-deploy users won't pre-create their datasets. I suggest deleting them.